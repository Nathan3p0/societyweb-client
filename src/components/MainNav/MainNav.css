.main__nav {
    display: flex;
    justify-content: space-between;
    padding: 1.5rem 2rem;
    background-color: #26575E;
}

.main__nav h1 {
    color: #fff;
    align-self: center;
}

.main__nav-login button {
    margin: 0 .5rem;
    padding: 12px 24px;
    border: none;
    background: #4ab3dd;
    color: #f2f2f2;
    text-transform: uppercase;
    letter-spacing: .09em;
    border-radius: 2px;
}

.main__nav-login button:hover {
    cursor: pointer;
    transform: scale(1.1);
    box-shadow: 1px 1px 5px #333333;
}