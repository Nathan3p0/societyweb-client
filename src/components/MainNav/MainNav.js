import React, { Component } from 'react';
import TokenService from '../../services/token-service';
import LoginInfoContext from '../../context/LoginInfoContext';
import './MainNav.css'

class MainNav extends Component {
    static contextType = LoginInfoContext;

    state = {}

    handleLogout = () => {
        this.context.clearStateOnLogout();
        TokenService.clearAuthToken();
    }

    renderLoginLink = () => {
        return (
            <div className="main__nav-login">
                <button onClick={this.context.leaderLoginToggle}>Leader Login</button>
                <button onClick={this.context.memberLoginToggle}>Member Login</button>
            </div>
        )
    }

    renderLogoutLink = () => {
        return (
            < div className="main__nav-logout" >
                <button onClick={this.handleLogout}>Logout</button>
            </div >
        )
    }
    render() {
        return (
            <nav className="main__nav">
                {!TokenService.hasAuthToken() ? <h1 className="main__nav-toggle" onClick={this.props.mainToggle}>SocietyWeb</h1> : <h1>SocietyWeb</h1>}
                {TokenService.hasAuthToken() ? this.renderLogoutLink() : this.renderLoginLink()}
            </nav>
        );
    }
}

export default MainNav;