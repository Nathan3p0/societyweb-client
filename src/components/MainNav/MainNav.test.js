import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import MainNav from './MainNav'

describe('Member Login Component Rendering Tests', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<MainNav />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders to UI as expected', () => {
        const tree = renderer.create(<MainNav />).toJSON();
        expect(tree).toMatchSnapshot()
    })
})