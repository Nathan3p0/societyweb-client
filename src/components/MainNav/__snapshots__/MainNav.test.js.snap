// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Member Login Component Rendering Tests renders to UI as expected 1`] = `
<nav
  className="main__nav"
>
  <h1>
    SocietyWeb
  </h1>
  <button>
    Leader Login
  </button>
  <button>
    Member Login
  </button>
  <button>
    Logout
  </button>
</nav>
`;
